File: Problems/0001_0099/0002_Add_Two_Number/Project_Java/Operate_ListNode.java
public class Operate_ListNode
{
    public ListNode set_ListNode(int[] nums)
    {
        return set_ListNode(nums, 0);
    }

    public ListNode set_ListNode(int[] nums, int index)
    {
        if (nums == null)
            return null;

        if (index >= nums.length)
            return null;
        
        ListNode node = new ListNode(nums[index]);
        node.next = set_ListNode(nums, index + 1);

        return node;
    }

    public String output_ListNode(ListNode node)
    {
        if (node == null)
            return "";

        String resultStr = Integer.toString(node.val);

        if (node.next != null)
            resultStr += " -> " + output_ListNode(node.next);

        return (resultStr);
    }
}

File: Problems/0001_0099/0025_Reverse_Nodes_in_k-Group/Project_Go/Operate_ListNode.go
package main

import "strconv"

func setListNode(nums []int) *ListNode {
	return setListNode2(nums, 0)
}

func setListNode2(nums []int, index int) *ListNode {
	if nums == nil {
		return nil
	}

	if index >= len(nums) {
		return nil
	}

	node := new(ListNode)
	node.Val = nums[index]
	node.Next = setListNode2(nums, index+1)

	return node
}

func outputListNode(node *ListNode) string {
	if node == nil {
		return ""
	}

	resultStr := strconv.Itoa(node.Val)

	if node.Next != nil {
		resultStr += " -> " + outputListNode(node.Next)
	}

	return (resultStr)
}
